src/ts/helpers/parse.ts

```diff
@@ -88,16 +96,40 @@ export function formatDateLocalized(date: Date): string {
   return `${date.toUTCString()}</br>(local time: ${date.toLocaleString()})`;
 }
 
-const BYTES_PER_KB = 1024;
-const BYTES_PER_MB = 1024 * BYTES_PER_KB;
+const bytesPerKb = 1024;
+const bytesPerMb = 1024 * bytesPerKb;
 
 export function formatBytes(bytes: number): string {
   const raw = `${bytes} bytes`;
-  if (bytes >= BYTES_PER_MB) {
-    return `${raw} (~${roundNumber(bytes / BYTES_PER_KB, 1)} MB)`;
+  if (bytes >= bytesPerMb) {
+    return `${raw} (~${roundNumber(bytes / bytesPerKb, 1)} MB)`;
   }
-  if (bytes >= BYTES_PER_KB) {
-    return `${raw} (~${roundNumber(bytes / BYTES_PER_KB, 0)} kB)`;
+  if (bytes >= bytesPerKb) {
+    return `${raw} (~${roundNumber(bytes / bytesPerKb, 0)} kB)`;
   }
   return raw;
 }
+
+/** HTML character to escape */
+const htmlCharMap = {
+  "\"": "&quot",
+  "&": "&amp",
+  "'": "&#039",
+  "<": "&lt",
+  ">": "&gt",
+};
+/**
+ * Reusable regex to escape HTML chars
+ * Combined to improve performance
+ */
+const htmlChars = new RegExp(Object.keys(htmlCharMap).join("|"), "g");
```

Comment:
I noticed the previous replace was quite slow on large `content` strings, since it went through it several timer, this way it's precompiled and only runs through it once.